src/screens/DirectMessagesScreen.tsx

```diff
@@ -134,36 +145,38 @@ export const DirectMessagesScreen = ({
       ): item is CreateAttachmentResponse => !item.loading;
 
       // In practice their won't be more than one message at a time
-      newMessages.map(async (m) => {
-        const isLastMessage = newMessages.indexOf(m) === newMessages.length - 1;
-
-        await mutateAsync({
-          userIds: users.map((user) => user.id),
-          post: {
-            message: m.text,
-            ...(isLastMessage
-              ? {
-                  attachmentsV2: pendingImages
-                    .filter(removeLoadingImages)
-                    .map((i) => ({
-                      ...i.attachment,
-                      url: i.uploadConfig.fileLocation.permanentUrl,
-                    })),
-                }
-              : {}),
-          },
-        });
-
-        if (isLastMessage) {
-          setPendingImages([]);
-        }
-      });
+      await Promise.all(
+        newMessages.map(async (m) => {
+          const isLastMessage =
+            newMessages.indexOf(m) === newMessages.length - 1;
+
+          await mutateAsync({
+            userIds: users.map((user) => user.id),
+            post: {
+              message: m.text,
+              ...(isLastMessage
```

Comment:
Just for my own knowledge and I should have asked this on the original PR, what is the reason attachments are only included if the message is the last message?

Reply:
It's just a quirk of the gifted chat interface to say that there could be multiple messages but for our use-case I believe there is only ever one message. I just wanted to make sure the attachments were only added once in case I was wrong about why there could be multiple messages passed to the callback.